includes/admin/class-wc-admin-addons.php

```diff
@@ -476,18 +479,46 @@ public static function output_featured_sections( $sections ) {
 		}
 	}
 
+	/**
+	 * Returns in-app-purchase URL params.
+	 */
+	public static function get_in_app_purchase_url_params() {
+		$back_admin_path = isset( $_SERVER['REQUEST_URI'] ) ? esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
+		return array(
+			'in-app-purchase-site'        => site_url(),
+			'in-app-purchase-back'        => $back_admin_path,
+			'in-app-purchase-woo-version' => WC_VERSION,
+		);
+	}
+
+	/**
+	 * Add in-app-purchase URL params to link.
+	 *
+	 * Adds various url parameters to a url to support a streamlined
+	 * flow for obtaining and setting up WooCommerce extensons.
+	 *
+	 * @param string $url    Destination URL.
+	 */
+	public static function add_in_app_purchase_url_params( $url ) {
+		return add_query_arg(
+			self::get_in_app_purchase_url_params(),
+			$url
+		);
+	}
+
 	/**
 	 * Outputs a button.
 	 *
-	 * @param string $url
-	 * @param string $text
-	 * @param string $theme
-	 * @param string $plugin
+	 * @param string $url    Destination URL.
+	 * @param string $text   Button label text.
+	 * @param string $theme  Button style class (not to be confused with WP theme!).
+	 * @param string $plugin The plugin the button is promoting.
 	 */
 	public static function output_button( $url, $text, $theme, $plugin = '' ) {
 		$theme = __( 'Free', 'woocommerce' ) === $text ? 'addons-button-outline-green' : $theme;
```

Comment:
Perhaps a good idea to rename this variable to something that represent it better as a css class instead of theme, which sounds confusing and makes you think it relates to an actual WP theme. I know the docblock gives the description but still confusing at first.

Reply:
Agreed, fixed here: f97fa34c016db49afc01dddedff667ec828d6a4b